File: frontend/src/services/userService.js
import apiClient from "./apiClient";

const unwrap = (response) => {
    return response.data?.data ?? response.data?.result ?? response.data;
};

export const userService = {
    getAllUsers: async (page = 0, size = 10) => {
        const res = await apiClient.get(`/users?page=${page}&size=${size}`);
        return res.data?.result ?? res.data?.data ?? res.data;
    },

    getGrowthStats: async () => {
        const res = await apiClient.get("/users/growth-stats");
        return res.data?.result ?? res.data?.data ?? res.data;
    },

    updateUser: async (id, payload) => {
        const res = await apiClient.put(`/users/${id}`, payload);
        return unwrap(res);
    },

    deleteUser: async (id) => {
        const res = await apiClient.delete(`/users/${id}`);
        return unwrap(res);
    },
};

File: backend/src/main/java/com/philosophy/rag/features/auth/service/UserService.java
package com.philosophy.rag.features.auth.service;

import com.philosophy.rag.features.auth.dto.UserGrowthStatsResponse;
import com.philosophy.rag.features.auth.dto.UserResponse;
import com.philosophy.rag.features.auth.dto.UserUpdateRequest;
import com.philosophy.rag.features.learning.dto.UserDashboardResponse;
import org.springframework.data.domain.Page;
import org.springframework.web.multipart.MultipartFile;

import java.util.UUID;

public interface UserService {

    UserResponse getUser(UUID userId);

    UserResponse getCurrentUser();

    UUID getCurrentUserId();

    Page<UserResponse> getAllUsers(int page, int size);

    UserResponse updateUser(UUID userId, UserUpdateRequest request);

    void deleteUser(UUID userId);

    UserResponse uploadAvatar(UUID userId, MultipartFile file);

    UserDashboardResponse getDashboardStats(String username);

    void completeFile(String username, String s3Key);

    UserGrowthStatsResponse getGrowthStats();
}
